packages/monitor-v2/src/monitor-og/SnapshotVerification.ts

```diff
@@ -417,6 +420,41 @@ const verifyRules = (parsedRules: RulesParameters, proposal: SnapshotProposalGra
   return { verified: true };
 };
 
+// Check if the proposal has been executed before.
+const hasBeenExecuted = async (
+  currentProposal: TransactionsProposedEvent,
+  params: MonitoringParams
+): Promise<boolean> => {
+  // Get all previous proposals with matching transactions and explanation for the same module that were proposed before
+  // the current proposal.
+  const og = await getOgByAddress(params, currentProposal.address);
+  const previousMatchingProposals = (
+    await runQueryFilter<TransactionsProposedEvent>(og, og.filters.TransactionsProposed(), {
+      start: 0,
+      end: currentProposal.blockNumber,
+    })
+  ).filter(
+    (previousProposal) =>
+      previousProposal.args.proposalHash === currentProposal.args.proposalHash &&
+      previousProposal.args.explanation === currentProposal.args.explanation &&
+      (previousProposal.blockNumber !== currentProposal.blockNumber ||
+        previousProposal.transactionIndex < currentProposal.transactionIndex ||
+        (previousProposal.transactionHash === currentProposal.transactionHash &&
+          previousProposal.logIndex < currentProposal.logIndex))
+  );
+
+  // Return true if any of the matching proposals have been executed.
+  const executedAssertionIds = (
```

Comment:
If we're searching to find executed ones already, why do we need to filter out the current proposal in the previous query?

Couldn't we just search for all proposals with matching txns and explanations, grab the assertionIds, then make sure none have been executed? The bit about checking the block number, tx hash, etc, seems unnecessary, no?

Reply:
Yes, this can be simplified. I pushed the last commit b2bf9472518b7d1b7e3e8245f4dd4acfbd156e23 to match proposals till the current proposals block not worrying to include the current proposal or any subsequent proposals in the same tx/block as they cannot be executed in the same block (liveness > 0)